pkg/core/annotations.go

```diff
@@ -35,3 +37,39 @@ func (c *client) bumpNonceAnnotation(s *v1alpha1.Service) {
 	annotations[buildAnnotation] = strconv.Itoa(i + 1)
 	s.Spec.RunLatest.Configuration.RevisionTemplate.SetAnnotations(annotations)
 }
+
+func (c *client) bumpNonceAnnotationForBuild(s *v1alpha1.Service) {
+	configurationSpec := s.Spec.RunLatest.Configuration
+	build := getBuild(configurationSpec)
+	if build != nil {
+		annotations := build.GetAnnotations()
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+		nonce := annotations[buildAnnotation]
+		i, err := strconv.Atoi(nonce)
+		if err != nil {
+			i = 0
+		}
+		annotations[buildAnnotation] = strconv.Itoa(i + 1)
+		build.SetAnnotations(annotations)
+		s.Spec.RunLatest.Configuration.Build = &v1alpha1.RawExtension{
+			Object: build,
+		}
+	}
+}
+
+func getBuild(configSpec v1alpha1.ConfigurationSpec) *unstructured.Unstructured {
+	// Trap panic from the Knative resources.GetBuild function call
```

Comment:
why does it panic? That seems like something that should be fixed upstream

Reply:
there are a couple of panics in the upstream code, seems to be when the data doesn't match what is expected

Reply:
I reworked our tests to create a more complete Build object to avoid triggering a panic